tools/shareable-examples/src/utils/get-example-module.ts

```diff
@@ -0,0 +1,163 @@
+      if (assignmentName === 'entryComponents') {
+        const entryComponentsNode = visitEachChild(
+          node,
+          visitSpraedDeclarationAndReplaceIt,
+          context,
+        );
+        return createPropertyAssignment(
+          'exports',
+          entryComponentsNode.initializer,
+        );
+      }
+      return node;
+    }
+
+    /** Visit an importDeclaration and remove imports if necessary. */
+    function visitImportDeclaration(node: ImportDeclaration): Node | undefined {
+      const importFrom = node.moduleSpecifier.getText().replace(/'/g, '');
+      // By definition, our examples are nested each within its own folder
+      // we will need to keep the import for this one example, and all other
+      // imports, that are not nested in a folder here.
+      if (
+        // Check if it is a relative import
+        importFrom.startsWith('.') &&
+        // Check if it is not the one import we actually want to keep
+        !importFrom.includes(basename(exampleRoot)) &&
+        // Check if it is not any of the other relative imports we want to ditch.
+        importFrom.split('/').length > 2
+      ) {
+        return undefined;
+      }
+      return node;
+    }
+
+    /** Main visit function. */
+    function visit(node: Node): Node | undefined {
+      // Remove all import declarations that are not needed anymore.
+      if (isImportDeclaration(node)) {
+        return visitNode(node, visitImportDeclaration);
+      }
+
+      // Remove the unnecessary array variable declaration
+      if (isVariableStatement(node)) {
+        // If the variable statement is the declarations array, remove it.
+        if (/DT_(.*?)_EXAMPLES/.test(node.getText())) {
+          return undefined;
+        }
+      }
+
+      // Replace the declarations and entryComponents assignments
+      if (isPropertyAssignment(node)) {
+        return visitNode(node, visitPropertyAssignment);
+      }
+      return visitEachChild(node, visit, context);
+    }
+    return node => visitNode(node, visit);
+  };
+  return transformer;
+}
+
+export async function getExampleModule(
+  exampleRoot: string,
+  exampleClassName: string,
+): Promise<ExampleAstFile> {
+  // Get the module file path
+  // This assumes that the module is one level up the folder structure
+  // and matches the pattern '.module.ts'
+  const moduleFilePath = readdirSync(resolve(exampleRoot, '../'))
+    .filter(file => file.endsWith('module.ts'))
+    .map(file => join('../', file))[0];
+  const moduleAstFile = await tsCreateSourceFile(
+    resolve(exampleRoot, moduleFilePath),
+  );
+
+  const result = transform(moduleAstFile, [
+    createTransformer(exampleRoot, exampleClassName),
+  ]);
+
+  // To apply the changes the transformed file needs to be passed
+  // to a printer to get the updated file.
+  const printer = createPrinter();
+  const content = printer.printFile(result.transformed[0]);
+  return {
```

Comment:
the ast has the path and the content inside it. So you can just return the sourceFile.

Reply:
I want  to keep that structure of ExampleFiles as it applies to ones that have an ast and ones that do not need one. I don't want to handle it differently when writing them to disk.